lisa/operating_system.py

```diff
@@ -150,36 +221,106 @@ def type_name(cls) -> str:
     def name_pattern(cls) -> Pattern[str]:
         return re.compile(f"^{cls.type_name()}$")
 
-    def _install_packages(self, packages: Union[List[str]]) -> None:
+    def _install_packages(
+        self, packages: Union[List[str]], signed: bool = True
+    ) -> None:
+        raise NotImplementedError()
+
+    def _package_exists(self, package: str, signed: bool = True) -> bool:
         raise NotImplementedError()
 
     def _initialize_package_installation(self) -> None:
         # sub os can override it, but it's optional
         pass
 
+    def _get_os_version(self) -> OsVersion:
+        os_version = OsVersion("")
+        # try to set OsVersion from info in /etc/os-release.
+        cmd_result = self._node.execute(cmd="cat /etc/os-release", no_error_log=True)
+        if cmd_result.exit_code == 0 and cmd_result.stdout != "":
+            for row in cmd_result.stdout.splitlines():
+                os_release_info = self.__os_info_pattern.match(row)
+                if os_release_info:
+                    if os_release_info.group("name") == "NAME":
+                        os_version.vendor = os_release_info.group("value")
+                    elif os_release_info.group("name") == "VERSION_ID":
+                        os_version.release = os_release_info.group("value")
+                    elif os_release_info.group("name") == "VERSION":
+                        os_version.codename = get_matched_str(
+                            os_release_info.group("value"),
+                            self.__distro_codename_pattern,
+                        )
+            if os_version.vendor == "":
+                raise LisaException("OS version information not found")
+        else:
+            raise LisaException(
+                "Error in running command 'cat /etc/os-release'"
+                f"stderr: {cmd_result.stderr}"
+            )
+
+        return os_version
+
+    # _resolve_package_name
+    # A package can be a string or a tool or a type of tool.
```

Comment:
The explanation of a method should be put into the method, and it may be generated to document in future. like below,
```python
    def _resolve_package_name(self, package: Union[str, Tool, Type[Tool]]) -> str:
        """
        Resolve it to a standard package_name so it can be installed.
        """
```

Reply:
Done as per you suggested.